pkg/monitor/notary_fetcher.go

```diff
@@ -0,0 +1,71 @@
+package monitor
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/gas"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/nep17"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+)
+
+type (
+	// NotaryFetcher allows to fetch notary balances from account.
+	NotaryFetcher struct {
+		cli   nep17.Invoker
+		cache *big.Float
+	}
+)
+
+// NewNotaryFetcher is a constructor for NotaryFetcher.
+func NewNotaryFetcher(cli nep17.Invoker) (*NotaryFetcher, error) {
+	return &NotaryFetcher{
+		cli: cli,
+	}, nil
+}
+
+func (b *NotaryFetcher) decimals() (*big.Float, error) {
+	if b.cache != nil {
+		return b.cache, nil
+	}
+
+	dec, err := gas.NewReader(b.cli).Decimals()
+	if err != nil {
+		return nil, err
+	}
+
+	b.cache = big.NewFloat(math.Pow10(dec))
+
+	return b.cache, nil
+}
+
+func (b *NotaryFetcher) format(balance *big.Int) (float64, error) {
+	multiplier, err := b.decimals()
+	if err != nil {
+		return 0, err
+	}
+
+	var bigFloat big.Float
+	bigFloat.SetInt(balance)
+	bigFloat.Quo(&bigFloat, multiplier)
+
+	res, _ := bigFloat.Float64()
+
+	return res, nil
+}
+
+// FetchNotary returns the notary balance of the given account.
+func (b *NotaryFetcher) FetchNotary(account util.Uint160) (float64, error) {
+	balance, err := gas.NewReader(b.cli).BalanceOf(account)
```

Comment:
`gas.NewReader` will just use GAS hash. We need to ask notary contract about it. Btw, there is https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/rpcclient/notary#ContractReader

Reply:
Sure. What about decimals is this case, I left 8 as a cost. Is it OK?

Reply:
It's GAS, so it's 8 (or `gas.Decimals()` technically).